packages/web-app-files/src/mixins/actions/restore.ts

```diff
@@ -43,7 +53,129 @@ export default {
     ...mapMutations('runtime/spaces', ['UPDATE_SPACE_FIELD']),
     ...mapMutations(['SET_QUOTA']),
 
-    async $_restore_trigger({ resources }) {
+    getParentFolderFromResource(resource) {
+      return resource.path.slice(0, resource.path.lastIndexOf(resource.name))
+    },
+    getWebdavParentFolderFromResource(resource) {
+      const parentPath = this.getParentFolderFromResource(resource)
+      return buildWebDavFilesPath(this.user.id, parentPath)
+    },
+    async collectRestoreConflicts(resources) {
+      const parentFolders = {}
+      const conflicts = []
+      const resolvedResources = []
+      for (const resource of resources) {
+        const webDavParentPath = this.getWebdavParentFolderFromResource(resource)
+
+        // ? check if parent folder has already been requested
+        let parentResources = []
+        if (webDavParentPath in parentFolders) {
+          parentResources = parentFolders[webDavParentPath]
+        } else {
+          try {
+            const listResponse = await this.$client.files.list(
```

Comment:
Could you please use the webdav client from clientService for new code?

Reply:
yes